src/Symfony/Component/Workflow/Tests/WorkflowTest.php

```diff
@@ -592,6 +592,32 @@ public function testGetEnabledTransitions()
         $this->assertSame('t5', $transitions[0]->getName());
     }
 
+    public function testGetEnabledTransition()
+    {
+        $definition = $this->createComplexWorkflowDefinition();
+        $subject = new Subject();
+        $eventDispatcher = new EventDispatcher();
```

Comment:
I don't think we need this complexity. Guard are already massively tested.
you can remove all things related to the Event Dispatcher + Guard?
It will make tests simpler to understand :) 
Thanks

Reply:
done !